### PR TITLE
Drug Molecule Generation with VAE to keras 3.0 (Tensorflow backend only)

### PR DESCRIPTION
**Pull request by chunduriv**
This PR changes the Drug Molecule Generation with VAE to keras 3.0 (Tensorflow backend only), as requested in https://github.com/keras-team/keras-cv/issues/2211

Please review the attached [gist](https://colab.sandbox.google.com/gist/chunduriv/d78bc46bbd763f926b576cfceb7e2150/molecule_generation_tf.ipynb).

### CODE REVIEW

File: examples/generative/molecule_generation.py [hunk 1]
```diff
@@ -70,14 +70,18 @@
...
 import ast
 
 import pandas as pd
 import numpy as np
 
 import tensorflow as tf
-from tensorflow import keras
-from tensorflow.keras import layers
+import keras
+from keras import layers, ops
```
Comment (fchollet) [inline examples/generative/molecule_generation.py:84]: Please use one line per import

File: examples/generative/molecule_generation.py [hunk 2]
```diff
@@ -374,12 +378,16 @@ def get_decoder(dense_units, dropout_rate, latent_dim, adjacency_shape, feature_
 
 
 class Sampling(layers.Layer):
+    def __init__(self, seed=None, **kwargs):
+        super().__init__(**kwargs)
+        self.seed_generator = seed or keras.random.SeedGenerator()
```
Comment (fchollet) [inline examples/generative/molecule_generation.py:383]: Use `self.seed_generator = keras.random.SeedGenerator(seed)`

File: examples/generative/molecule_generation.py [hunk 3]
```diff
@@ -409,20 +417,21 @@ def call(self, inputs):
...
 
 class MoleculeGenerator(keras.Model):
-    def __init__(self, encoder, decoder, max_len, **kwargs):
+    def __init__(self, encoder, decoder, max_len, seed=None, **kwargs):
         super().__init__(**kwargs)
         self.encoder = encoder
         self.decoder = decoder
         self.property_prediction_layer = layers.Dense(1)
         self.max_len = max_len
+        self.seed_generator = seed or keras.random.SeedGenerator()
```
Comment (fchollet) [inline examples/generative/molecule_generation.py:426]: Same here

File: examples/generative/molecule_generation.py [hunk 4]
```diff
@@ -492,32 +505,34 @@ def _gradient_penalty(self, graph_real, graph_generated):
...
+        features = ops.one_hot(features, num_classes=ATOM_DIM, axis=2)
         return [
             graph_to_molecule([adjacency[i].numpy(), features[i].numpy()])
             for i in range(batch_size)
         ]
 
     def call(self, inputs):
         z_mean, log_var = self.encoder(inputs)
-        z = Sampling()([z_mean, log_var])
+        z = Sampling(seed=self.seed_generator)([z_mean, log_var])
```
Comment (fchollet) [inline examples/generative/molecule_generation.py:535]: Do not create a layer inside `call`, instead create it in `__init__` and call it here

**Review by fchollet — COMMENTED** [review]
Thanks for the PR -- nice work!

File: examples/generative/molecule_generation.py (hunk 1 above)
Comment (chunduriv) [reply examples/generative/molecule_generation.py]: Done

File: examples/generative/molecule_generation.py (hunk 2 above)
Comment (chunduriv) [reply examples/generative/molecule_generation.py]: Done

File: examples/generative/molecule_generation.py (hunk 3 above)
Comment (chunduriv) [reply examples/generative/molecule_generation.py]: Done

File: examples/generative/molecule_generation.py (hunk 4 above)
Comment (chunduriv) [reply examples/generative/molecule_generation.py]: Done

**Review by fchollet — COMMENTED** [review]
Looking great -- please add the generated files and update `scripts/examples_master.py`.

File: examples/generative/molecule_generation.py [hunk 5]
```diff
@@ -374,12 +379,16 @@ def get_decoder(dense_units, dropout_rate, latent_dim, adjacency_shape, feature_
...
+        super().__init__(**kwargs)
+        self.seed_generator = keras.random.SeedGenerator(seed)
+
     def call(self, inputs):
         z_mean, z_log_var = inputs
-        batch = tf.shape(z_log_var)[0]
-        dim = tf.shape(z_log_var)[1]
-        epsilon = tf.keras.backend.random_normal(shape=(batch, dim))
-        return z_mean + tf.exp(0.5 * z_log_var) * epsilon
+        batch = ops.shape(z_log_var)[0]
```
Comment (mattdangerw) [inline examples/generative/molecule_generation.py:388]: could do this in one line `batch, dim = ops.shape(z_log_var)` or `batch, dim, _ = ops.shape(z_log_var)` if it's higher rank

**Review by mattdangerw — APPROVED** [review]
lgtm!
